ui/appui-react/src/appui-react/widgets/WidgetDef.tsx

```diff
@@ -543,10 +547,82 @@ export class WidgetDef {
     if (!state) return;
     if (!frontstageDef.findWidgetDef(this.id)) return;
 
-    frontstageDef.dispatch({
-      type: "WIDGET_TAB_SHOW",
-      id: this.id,
-    });
+    const tabLocation = getTabLocation(state, this.id);
+    if (tabLocation && isPopoutTabLocation(tabLocation)) {
+      const testWindow = UiFramework.childWindows.find(
+        tabLocation.popoutWidgetId
+      );
+      if (testWindow) {
+        // We have to check if Safari is the browser being used. Safari has security
+        // measures that do not allow you to change focus to another tab programmatically.
+        // Therefore the only and recommended way around this at the moment, is to close
+        // and then open the tab again. This remounts the popout. Thats why this will not be done,
+        // because we don't want the state of the widget to be lost. Instead a toast will
+        // be displayed that informs the user that safari cannot focus the popout.
+        // Apple support docs: https://discussions.apple.com/thread/251676767?sortBy=best
+        const isSafari =
+          navigator.userAgent.toLowerCase().indexOf("safari/") > -1;
+
+        const isChrome =
+          navigator.userAgent.toLowerCase().indexOf("chrome") > -1;
+
+        if (
+          !isChrome &&
+          isSafari &&
+          window.localStorage.getItem("hideSafariPopoutFocusMessage") !== "true"
+        ) {
+          const checkbox = (
+            <div>
+              <input
+                type="checkbox"
+                id="doNotShowAgain"
+                name="doNotShowAgain"
+                onChange={() => {
+                  if (
+                    window.localStorage.getItem(
+                      "hideSafariPopoutFocusMessage"
+                    ) !== "true"
+                  ) {
+                    window.localStorage.setItem(
+                      "hideSafariPopoutFocusMessage",
+                      "true"
+                    );
+                  } else {
+                    window.localStorage.setItem(
+                      "hideSafariPopoutFocusMessage",
+                      "false"
+                    );
```

Comment:
Since setting this to `true` once will make the notification disappear forever, use `UiFramework.getUiStateStorage()` instead of `window.localStorage`, so if we need to change this later to a SettingsProvider and allow the user to change this preference, it will be easier.

Reply:
Changed to use UiStateStorage